usb/libloader.py

```diff
@@ -103,17 +105,17 @@ def locate_library(candidates, find_library=ctypes.util.find_library):
 
         libname = find_library(candidate)
         if libname:
-            return libname
+            liblist.append(libname)
 
         # On Apple Silicon, also check in `/opt/homebrew/lib`: homebrew patches
         # the Python interpreters it distributes to check that directory, but
         # other/stock interpreters don't know about it
         if use_apple_silicon_workaround:
             libname = "/opt/homebrew/lib/" + candidate + ".dylib"
             if os.path.isfile(libname):
-                return libname
-
-    return None
+                liblist.append(libname)
+        
+    return liblist
```

Comment:
We can't change the return type of the existing API, as it would be a breaking change.

For a new API, a generator that `yield`s candidates might be a cleaner way to do it. But because choices like these, in a library as mature as this one, I would also prefer that the new API be internal only in the beginning.

That said, it should be fairly simple to re-implement the old, public, API on top of the new, possibly generator-based, one (avoiding duplication).